### PR TITLE
Refactor backend classes and remove deprecated methods in PyTorch backends

### PR DESCRIPTION
**Pull request by TaekyungHeo**
## Summary
- Renamed backendFunctions to BaseBackend and refactored associated modules for improved abstraction.
- Removed getBusBW from BaseBackend.
- Removed initialize_tcpstore and UCC plugin support from PyTorchDistBackend.
- Reorganized module structure by relocating files to a backend directory.

## Test Plan
* Note: The following command was executed using https://github.com/facebookresearch/param/pull/148.
```
$ comm_replay --trace-type et --trace-path /home/sanshang/021_debug/000_code/param/trace/traces_megatronlm_gpt_43B_32ranks_pytnightly0703/execution_trace
```
image

### CODE REVIEW

File: et_replay/comm/backend/pytorch_dist_backend.py [hunk 1]
```diff
@@ -996,18 +998,12 @@ def __init__(self, bootstrap_info, commsParams):
             if isinstance(self.commsParams, dict)
             else self.commsParams.backend
         )
-        # Import ucc plugin
-        if backend == "ucc":
-            # try OSS/setup.py
+        # Import Fairring
+        if backend == "fairring":
```
Comment (shengfukevin) [inline et_replay/comm/backend/pytorch_dist_backend.py:1002]: Looks like "fairring" is the supported backend. Shall we add "fairring" into supportedC10dBackends in base_backend.py? and remove "ucc"?

**Comment by shengfukevin** [comment]
Thanks for the PR. Looks good to me. Have one minor inline comment, please check.

**Comment by shengfukevin** [comment]
@TaekyungHeo, I saw quite a few BLACK lint error, does our github project enable BLACK already?

**Comment by TaekyungHeo** [comment]
@shengfukevin, this repository has a linter, as you can see at https://github.com/facebookresearch/param/blob/d9eda8d2c7f43fda2304e04d5ba99a9db1d5c37d/.github/workflows/python_lint.yml#L22. We ensured that our PRs pass the black linter as described in the GitHub pipeline. Do you see any differences between these rules and Meta’s internal rules? If so, what would be the best solution? I suggest updating the Param Black rules to align with Meta’s Black rules.

**Comment by TaekyungHeo** [comment]
@shengfukevin Do you have any suggestions for NVIDIA to avoid linter errors in advance?

**Comment by shengfukevin** [comment]
> @shengfukevin Do you have any suggestions for NVIDIA to avoid linter errors in advance?

I am asking PyTorch team and see how they make user OSS linter aligned with internal linter.

File: et_replay/comm/backend/pytorch_dist_backend.py (hunk 1 above)
Comment (GSSBMW) [reply et_replay/comm/backend/pytorch_dist_backend.py:1002]: I think so. Will try to update. Thanks!
@Sergei-Lebedev

**Comment by shengfukevin** [comment]
@TaekyungHeo @GSSBMW, I fixed the meta lint issue after I imported this PR to meta. However, I can not export the changed code to this PR since I am not the owner. So I created a new PR https://github.com/facebookresearch/param/pull/149 which include the lint fix. I also changed supportedC10dBackends in base_backend.py to add fairring and remove ucc. Please review https://github.com/facebookresearch/param/pull/149.

**Comment by TaekyungHeo** [comment]
Closing this PR as @shengfukevin has created a new one to fix lint errors: https://github.com/facebookresearch/param/pull/149

**Pull request closed without merging**